Create Chips component

StheffanyHadlich opened the pull request:
**Component**: Chips

**Description**: implement chips component

**Design**:
![image](https://user-images.githubusercontent.com/31314523/66179418-48c97080-e63f-11e9-931b-02dc078e30b5.png)

This component accepts an icon, text, an button icon and styles variations according to pre defined themes.

**Breaking Changes**:

- [] Breaking Changes


---
Closes: #82

Commit:
Create Chips component

bert-e commented:
# Hello stheffanyhadlich,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

carlito-scality commented on src/lib/components/chips/Chips.component.js:
```diff
@@ -0,0 +1,114 @@
+    const brandingTheme = mergeTheme(props.theme, defaultTheme);
+
+    const brandLight = lighten(0.1, brandingTheme[props.variant]).toString();
+    console.log(brandingTheme);
```
It should be removed.

carlito-scality commented on src/lib/components/chips/Chips.component.js:
```diff
@@ -0,0 +1,114 @@
+  onClick, 
+  color = defaultTheme.white,
+}: Props) {
+  return ( //tirar negação dos ícone
```
Comments should be in english 😉

StheffanyHadlich replied:
oh sorry, my bad

carlito-scality commented on src/lib/components/chips/Chips.component.js:
```diff
@@ -0,0 +1,114 @@
+  variant?: Variant,
+  icon?: Node,
+  buttonIcon?: Node,
+  onClick?: any => void,
```
It would be nicer to have a remove button as part of this component as you do in the storybook. 
We can add ‘onRemove?: any => void’ in the Props. Then if ‘onRemove’ is defined we display the remove Button, otherwise not.

carlito-scality commented on src/lib/components/chips/Chips.component.js:
```diff
@@ -0,0 +1,114 @@
+
+function Chips({
+  text,
+  variant,
```
You can set the default value here instead of putting it in the defaulProps like for ´color’

MonPote replied:
Yes, and `defaultProps` for function will eventually [be deprecated](https://github.com/reactjs/rfcs/pull/107)

carlito-scality commented on src/lib/components/chips/Chips.component.js:
```diff
@@ -0,0 +1,114 @@
+
+type Props = {
+  text: string,
+  color?: String,
```
string instead of String

carlito-scality reviewed (changes requested):
Hello! I have some small request changes but very good job. Thanks 😊

carlito-scality commented on src/lib/components/chips/Chips.component.js:
```diff
@@ -0,0 +1,114 @@
+      background-color: ${brandingTheme[props.variant]};
+      color: ${props.color};
+    
+      &:hover {
```
hover and active should be needed only when onClick is defined. 
« pointer: cursor » should be added to the hover as well

StheffanyHadlich replied:
Can I add those when onRemove is defined as well?

carlito-scality replied:
I think the Remove button should have its own hover style so that in the case when we have onRemove and onClick both defined, we know what is clicked when clicking.

StheffanyHadlich commented:
> Hello! I have some small request changes but very good job. Thanks

haha thanks, I'll fix it

carlito-scality commented:
It would be nicer to have this component with different sizes but we could implement this feature later in another PR :)

Commit:
Fix styles

carlito-scality committed:
Add missing type "base" in Variant

carlito-scality committed:
Chips: improve styles
- Add missing import relative to Flow
- Improve css
- Remove defaultProps
- Update tests

carlito-scality commented:
@StheffanyHadlich I just pushed some small changes. If you agree, we can merge this PR :)

StheffanyHadlich commented:
> @StheffanyHadlich I just pushed some small changes. If you agree, we can merge this PR :)

That is fine with me :)
